lib/intercept.js

```diff
@@ -146,32 +146,29 @@ function interceptorsFor(options) {
 
   debug('filtering interceptors for basepath', basePath)
 
-  //  First try to use filteringScope if any of the interceptors has it defined.
-  let matchingInterceptor
-  _.each(allInterceptors, function(interceptor) {
-    _.each(interceptor.scopes, function(scope) {
-      const { filteringScope } = scope.__nock_scopeOptions
+  // First try to use filteringScope if any of the interceptors has it defined.
+  for (const { key, interceptors, allowUnmocked } of Object.values(
+    allInterceptors
+  )) {
+    for (const interceptor of interceptors) {
+      const { filteringScope } = interceptor.__nock_scopeOptions
 
-      //  If scope filtering function is defined and returns a truthy value
-      //  then we have to treat this as a match.
+      // If scope filtering function is defined and returns a truthy value then
+      // we have to treat this as a match.
       if (filteringScope && filteringScope(basePath)) {
         debug('found matching scope interceptor')
 
-        //  Keep the filtered scope (its key) to signal the rest of the module
-        //  that this wasn't an exact but filtered match.
-        scope.__nock_filteredScope = scope.__nock_scopeKey
-        matchingInterceptor = interceptor.scopes
-        //  Break out of _.each for scopes.
-        return false
+        // Keep the filtered scope (its key) to signal the rest of the module
+        // that this wasn't an exact but filtered match.
+        interceptor.__nock_filteredScope = interceptor.__nock_scopeKey
```

Comment:
Confusing but I think correct ^^, as the old `interceptor` variable was actually one of these `{ key, interceptors, allowUnmocked } `structures, and `scope` was actually an Interceptor.